code/modules/mob/living/carbon/human/death.dm

```diff
@@ -47,12 +47,18 @@
 		ticker.mode.update_cult_icons_removed(mind)
 		ticker.mode.cult -= mind
 
-	species = null
+	if(species)
+		qdel(species)
+		species = null
```

Comment:
It might be different from Bay but isn't species a singleton datum shared by all relevant mob?